.github/workflows/go-test.yml

```diff
@@ -0,0 +1,26 @@
+name: Go tests
+on: [push]
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - windows-latest
+          - ubuntu-latest
+        go-version:
+          - 1.18.x
+          - 1.22.x
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Configure git in Windows
+        run: |
+          git config --global core.autocrlf input
+        if: ${{ matrix.os == 'windows-latest' }}
+      - uses: actions/checkout@v4
+      - name: Setup Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v4
```

Comment:
There is a newer version of this action (`setup-go@v5`), it could be tried with that version